### PR TITLE
Front end and data binding for instructor create survey

### PR DESCRIPTION
**Pull request by jepetty**
Doesn't necessarily need to be merged, but wanted input on how I've set up the different data structures and any input on the look of it

Todo: 
- make separate project creation page for instructors
- hook this up with the student side
- make data persistent

### CODE REVIEW

**Comment by jepetty** [comment]
@haydenreich I'm looking at your PR and I'm thinking you should let me know if I need to change some of the things from "weights" to "should be considered: true or false" to make it easier to parse on your end

File: hatServer/static/js/services/surveyQuestions.js [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+    var fifthComment = {
+        included: true,
+        required: false
+    };
+    var ipPreference = {
+        included: true,
+        required: true,
+        weight: 1
+    };
+    var ipOptions = {
```
Comment (haydenreich) [inline hatServer/static/js/services/surveyQuestions.js]: Variables for ip rights I used were RETAIN and NO_PREF if you want to keep in consistent, or I can change it in the algorithm.
I know 'Depends on the project' was an option, but it's kind of useless as far as the algorithm goes, so I was just going to make it NO_PREF as well

File: hatServer/static/js/services/surveyQuestions.js [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+        dependsIncluded: true,
+        noPref: 'I don\'t have a preference to retain IP rights to my work',
+        noPrefIncluded: true
+    };
+    var leadership = {
+        included: true,
+        required: true,
+        weight: 1
+    };
+    var leadershipOptions = {
```
Comment (haydenreich) [inline hatServer/static/js/services/surveyQuestions.js]: Same kind of deal here, STRONG_FOLLOW, PREFER_FOLLOW, NO_PREF, PREFER_LEAD, STRONG_LEAD. These were going to be enums, hence the caps...

File: hatServer/static/js/services/surveyQuestions.js [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+    };
+    var overallGPA = {
+        included: true,
+        required: true,
+        weight: 1
+    };
+    var csGPA = {
+        included: true,
+        required: true,
+        weight: 1
```
Comment (haydenreich) [inline hatServer/static/js/services/surveyQuestions.js]: Are these weights passed in the JSON object? I'm not sure I understand how they're used here, or do these values get input by the instructors when they determine how to weight variables? Either way, it'd probably be easier if the weight was passed in a separate interface and could just be loaded into the algorithm variable.cfg file

File: hatServer/static/js/services/surveyQuestions.js [hunk 4]
```diff
@@ -0,0 +1,461 @@
+'use strict';
+
+sortingApp.service('surveyQuestions', function () {
+    var surveyName = "";
+    var surveyDescription = 'HELLO JESSICA YOU DA BEST';
```
Comment (haydenreich) [inline hatServer/static/js/services/surveyQuestions.js:5]: Descriptive

File: hatServer/static/js/controllers/createSurvey.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    $scope.fifthComment = surveyQuestions.getFifthComment();
+    $scope.ipPreference = surveyQuestions.getIpPreference();
+    $scope.ipOptions = surveyQuestions.getIpOptions();
+    $scope.leadership = surveyQuestions.getLeadership();
+    $scope.leadershipOptions = surveyQuestions.getLeadershipOptions();
+    $scope.preferredPartners = surveyQuestions.getPreferredPartners();
+    $scope.bannedPartners = surveyQuestions.getBannedPartners();
+
+    $scope.createSurvey = function (form) {
+        console.log('creating dat survey do');
```
Comment (haydenreich) [inline hatServer/static/js/controllers/createSurvey.js]: how bout dat

File: hatServer/static/js/services/surveyQuestions.js [hunk 6]
```diff
@@ -0,0 +1,461 @@
...
+    var desiredSkills = ['Java', 'Python', 'PHP', 'C/C++', 'Mobile App Development (Android/iOS)', 'Web Applications', 'Embedded Systems', 'Database (MySQL, SQL, etc.)', 'User Interface/Experience', 'Statistics', 'Networking',
+        'Robotics', 'Computer Vision', 'Algorithms', 'Machine Learning'];
```
Comment (haydenreich) [inline hatServer/static/js/services/surveyQuestions.js]: I don't know if there's a convenient way to do this, but we need to make sure these skills are consistent with the list the instructors use when building the groups.

File: hatServer/static/js/controllers/createSurvey.js (hunk 5 above)
Comment (jeor0980) [reply hatServer/static/js/controllers/createSurvey.js]: cash me ousside

**Pull request closed without merging**